core/src/main/java/org/mskcc/cbio/portal/dao/DaoMutation.java

```diff
@@ -1602,4 +1602,58 @@ public static boolean hasDuplicateMutationEvents() throws DaoException {
         }
         return !(sqlQueryResult > 0);
     }
+
+    public static void createStructuralVariantData(int cancerStudyId) throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+
+        String[] statements = {
+            "INSERT INTO genetic_profile (STABLE_ID, CANCER_STUDY_ID, GENETIC_ALTERATION_TYPE, DATATYPE, NAME, DESCRIPTION, SHOW_PROFILE_IN_ANALYSIS_TAB) " +
+            "SELECT DISTINCT CONCAT(CANCER_STUDY_IDENTIFIER, '_structural_variants'), cancer_study.CANCER_STUDY_ID, 'STRUCTURAL_VARIANT','SV','Fusions','Fusions',0 " +
+            "FROM `mutation_event` " +
+            "JOIN `mutation` ON `mutation`.MUTATION_EVENT_ID = `mutation_event`.MUTATION_EVENT_ID " +
+            "JOIN `genetic_profile` ON `genetic_profile`.GENETIC_PROFILE_ID = `mutation`.GENETIC_PROFILE_ID " +
+            "JOIN `cancer_study` ON `cancer_study`.CANCER_STUDY_ID = `genetic_profile`.CANCER_STUDY_ID " +
+            "WHERE MUTATION_TYPE = 'fusion' " +
+            "AND NOT EXISTS (SELECT * FROM genetic_profile WHERE STABLE_ID=CONCAT(CANCER_STUDY_IDENTIFIER, '_structural_variants') AND CANCER_STUDY_ID = cancer_study.CANCER_STUDY_ID)" +
+            "AND cancer_study.CANCER_STUDY_ID = ?;",
```

Comment:
insert a new genetic_profile that should be delegated to the GeneticProfileReader dao object.